src/library/Pagination/Pages.js

```diff
@@ -0,0 +1,156 @@
+import { type Messages, componentTheme } from './Pagination';
+
+type Props = {
+  currentPage: number,
+  handleClick: (currentPage: number) => void,
+  handleIncrement: (incrementForward: boolean) => void,
+  hidePageNumbers?: boolean,
+  messages: Messages,
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  totalPages: number,
+  visibleRange: number
+};
+
+const Container = createStyledComponent(FlexItem, ({ theme: baseTheme }) => {
+  const theme = componentTheme(baseTheme);
+  return {
+    '& > button, & > span': {
+      '&:not(:last-child)': {
+        marginRight: theme.Pagination_gutterWidth
+      }
+    }
+  };
+});
+
+const firstPage = (current) => current === 1;
+const lastPage = (current, total) => current === total;
+
+const EllipsisButton = createThemedComponent(Button, ({ theme }) => ({
+  color_disabled: theme.color_theme
+}));
+
+const Buttons = ({
+  currentPage,
+  handleClick,
+  size,
+  totalPages,
+  visibleRange
+}: Props) => {
+  return [...Array(totalPages)]
+    .map((_, i) => i)
+    .map((_, index) => {
+      const page = index + 1;
+      const evenOffset = visibleRange % 2 === 0 ? 1 : 0;
+      let pagesBuffer;
+      if (
+        currentPage <= visibleRange ||
+        (visibleRange === 1 && currentPage === 2)
+      ) {
+        pagesBuffer = visibleRange - currentPage + 3 - evenOffset;
+      } else if (
+        currentPage > totalPages - visibleRange ||
+        (visibleRange === 1 && currentPage === totalPages - 1)
+      ) {
+        pagesBuffer =
+          visibleRange - (totalPages - currentPage) + 2 - evenOffset;
+      } else {
+        pagesBuffer = Math.ceil(visibleRange / 2);
+      }
+
+      const firstPageInRange =
+        page === currentPage - pagesBuffer && !firstPage(page);
+      const lastPageInRange =
+        page === currentPage + pagesBuffer && !lastPage(page, totalPages);
+      const isPageOutOfRange =
+        page < currentPage - pagesBuffer || page > currentPage + pagesBuffer;
+
+      let pageView = null;
+      if (firstPageInRange || lastPageInRange) {
+        pageView = (
+          <EllipsisButton
+            disabled
+            element="span"
+            key={page}
+            minimal
+            size="medium">
+            …
+          </EllipsisButton>
+        );
+      } else if (
+        !isPageOutOfRange ||
+        firstPage(page) ||
+        lastPage(page, totalPages)
+      ) {
+        pageView = (
+          <Button
+            minimal
+            primary={currentPage === page}
+            key={page}
+            onClick={handleClick.bind(null, page)}
+            size={size}>
+            {page}
+          </Button>
+        );
+      }
+      return pageView;
+    })
+    .filter((page) => !!page);
+};
+
+const GenericIncrementButton = ({
+  currentPage,
+  direction,
+  handleIncrement,
+  messages,
+  size,
+  totalPages
+}: Props & { direction: string }) => {
+  const incrementForward = direction === 'next' ? true : false;
```

Comment:
Might make sense to do something like `const next = direction === 'next'` and reference that throughout.